From 2e1d8fc5fa0690c6bdcadc4eae03d6c0706a998a Mon Sep 17 00:00:00 2001
From: adam wulw <adam.wulw@intel.com>
Date: Thu, 21 Feb 2019 12:26:40 +0100
Subject: [PATCH] ASoC: Intel: Skylake: Increase ROM_INIT_TIMEOUT to 100ms

In some platforms 70ms for ROM_INIT_TIMEOUT could be not enough.

Change-Id: I2f2557728f85102347565c6c1972f6e69daf4404
Signed-off-by: adam wulw <adam.wulw@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/38677
Reviewed-by: Slawinski, AmadeuszX <amadeuszx.slawinski@intel.com>
Tested-by: gkblditp <gkblditp@intel.com>
---
 sound/soc/intel/skylake/bxt-sst.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/sound/soc/intel/skylake/bxt-sst.c b/sound/soc/intel/skylake/bxt-sst.c
index 0561af1..77eaf01 100755
--- a/sound/soc/intel/skylake/bxt-sst.c
+++ b/sound/soc/intel/skylake/bxt-sst.c
@@ -26,7 +26,7 @@
 
 #define BXT_BASEFW_TIMEOUT	3000
 #define BXT_INIT_TIMEOUT	300
-#define BXT_ROM_INIT_TIMEOUT	70
+#define BXT_ROM_INIT_TIMEOUT	100
 #define BXT_IPC_PURGE_FW	0x01004000
 
 #define BXT_ROM_INIT		0x5
@@ -59,7 +59,7 @@
 /* Delay before scheduling D0i3 entry */
 #define BXT_D0I3_DELAY 5000
 
-#define BXT_FW_INIT_RETRY 10
+#define BXT_FW_INIT_RETRY 20
 
 #define GET_SSP_BASE(N)	(N > 4 ? 0x2000 : 0x4000)
 
@@ -336,19 +336,20 @@ static int bxt_load_base_firmware(struct sst_dsp *ctx)
 	for (i = 0; i < BXT_FW_INIT_RETRY; i++) {
 		ret = sst_bxt_prepare_fw(ctx, stripped_fw.data, stripped_fw.size);
 		if (ret < 0) {
-			dev_err(ctx->dev, "Error code=0x%x: FW status=0x%x\n",
+			dev_dbg(ctx->dev,
+				"Iteration %d Core En/ROM load failed: %d\nError code=0x%x: FW status=0x%x\n",
+				i, ret,
 				sst_dsp_shim_read(ctx, BXT_ADSP_ERROR_CODE),
 				sst_dsp_shim_read(ctx, BXT_ADSP_FW_STATUS));
-
-			dev_err(ctx->dev, "Iteration %d Core En/ROM load fail:%d\n", i, ret);
 			continue;
 		}
 		dev_dbg(ctx->dev, "Iteration %d ROM load Success:%d\n", i, ret);
 
 		ret = sst_transfer_fw_host_dma(ctx);
 		if (ret < 0) {
-			dev_err(ctx->dev, "Iteration %d Transfer firmware failed %d\n", i, ret);
-			dev_info(ctx->dev, "Error code=0x%x: FW status=0x%x\n",
+			dev_dbg(ctx->dev,
+			"Iteration %d Transfer firmware failed: %d\nError code=0x%x: FW status=0x%x\n",
+			i, ret,
 				sst_dsp_shim_read(ctx, BXT_ADSP_ERROR_CODE),
 				sst_dsp_shim_read(ctx, BXT_ADSP_FW_STATUS));
 
-- 
1.9.1

